### PR TITLE
Deprecate @python_class decorator

### PR DESCRIPTION
**Pull request by patrickod**
Deprecate `@python_class` and update affected unit tests to consume a
new `register_classes` fixture which performs the same work to register
classes for tests.

Update all Github action uses of Python 3.6 to Python 3.7 after it 3.6 went End of Life in December 23rd 2021.

PR checklist:

- [x] Added changelog entry.

### CODE REVIEW

**Review by dhatch — APPROVED** [review]
Needs a changelog entry, besides that LGTM.

**Review by dhatch — CHANGES_REQUESTED** [review]
Just realized the decorator is still referenced in places in our docs. For example here:

```
(venv) ➜  oso git:(patrickod/deprecate-python-polar-class-decorator) ✗ git --no-pager grep 'polar_class' -- docs
docs/content/any/project/changelogs/2020-04-24.md:#### `@external_class` decorator is now named `@polar_class`
docs/content/any/project/changelogs/2020-04-24.md:The `oso.polar_class()` decorator replaces `external_class`. The
docs/content/any/project/changelogs/2020-05-11.md:#### Removed arguments to `@polar_class` decorator
docs/content/any/project/changelogs/2020-05-11.md:The `oso.polar_class()` decorator no longer accepts `fields` or
docs/content/any/project/changelogs/2020-05-11.md:@polar_class(fields=("id", "name", "specialty"), methods=("department", "hospital"))
docs/content/any/project/changelogs/2020-05-11.md:@polar_class
docs/content/any/project/changelogs/NEXT.md:##### `@polar_class` is deprecated in favour of `Oso#register_class`
docs/content/any/project/changelogs/NEXT.md:The `@polar_class` decorator used to register classes with Polar has been deprecated. To register a class with Polar it is now necessary to use the [`Oso#register_class`](https://docs.osohq.com/reference/api/index.html#oso.Oso.register_class) API.
docs/content/python/guides/policies/data/data.md:  `Oso.polar_class()` decorator:
docs/content/python/reference/polar/classes.md:using either the `register_class()` method or the `polar_class()` decorator. An
docs/content/rust/guides/policies/data/data.md:      User::get_polar_class_builder()
docs/content/rust/guides/policies/data/data.md:     User::get_polar_class_builder()
docs/content/rust/guides/policies/data/data.md:  oso.register_class(User::get_polar_class())?;
docs/content/rust/guides/policies/data/data.md:  oso.register_class(User::get_polar_class())?;
docs/content/rust/reference/polar/classes.md:oso.register_class(User::get_polar_class())?;
docs/content/rust/reference/polar/classes.md:oso.register_class(User::get_polar_class())?;
docs/content/rust/reference/polar/classes.md:oso.register_class(User::get_polar_class())?;
docs/content/rust/reference/polar/classes.md:    User::get_polar_class_builder()
docs/content/rust/reference/polar/classes.md:    User::get_polar_class_builder()
docs/examples/context/python/02-context.py:from oso import polar_class
docs/examples/context/python/02-context.py:@polar_class
docs/examples/inheritance/python/inheritance_external.py:@oso.polar_class
docs/examples/inheritance/python/inheritance_external.py:@oso.polar_class
docs/examples/inheritance/python/inheritance_external.py:@oso.polar_class
docs/examples/inheritance/python/inheritance_external.py:@oso.polar_class
docs/examples/inheritance/python/inheritance_external.py:@oso.polar_class
docs/examples/user_types/python/01-user_classes.py:from oso import polar_class
docs/examples/user_types/python/01-user_classes.py:@polar_class
docs/examples/user_types/python/01-user_classes.py:@polar_class
docs/examples/user_types/python/02-user_classes.py:from oso import polar_class
docs/examples/user_types/python/02-user_classes.py:@polar_class
docs/examples/user_types/python/02-user_classes.py:@polar_class
```

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -172,10 +172,10 @@ jobs:
...
           override: true
           components: rustfmt, clippy
-      - name: Install Python 3.6
+      - name: Install Python 3.7
         uses: actions/setup-python@v1
         with:
-          python-version: "3.6"
+          python-version: "3.7"
       - name: Test python
         run: make python-test
```
Comment (gj) [inline .github/workflows/test.yml:180]: I'm not sure we should bump this version — we generally run tests against the minimum version of $lang that we support, regardless of whether that version is EOL.

**Review by gj — DISMISSED** [review]
1 question to resolve

File: .github/workflows/test.yml (hunk 1 above)
Comment (patrickod) [reply .github/workflows/test.yml]: updated back to `3.6`

**Review by dhatch — APPROVED** [review]

**Pull request merged**